presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -122,19 +123,19 @@ public Expression rewrite(FieldOrExpression fieldOrExpression)
         }
     }
 
-    public void put(Expression expression, Symbol symbol)
+    public void put(Expression expression, Symbol symbol, Collection<Expression> columnReferences)
```

Comment:
Looks like my comment on the previous incarnation of this commit got lost. Here it goes again:


> Can you explain how this is supposed to work?

> The `put` method used to mean: map `expression` to the translated symbol `symbol` so that anywhere `expression` used to appear, it should be replaced with the given symbol. E.g., if computing an aggregation like `sum(a * b)`, a pre-projection step would first compute `x = a * b` and the mapping `a * b => x` would be recorded in the translation map. The aggregation step would rewrite the expression `sum(a * b)` using the translations so that it cam be computed in terms of the output of the pre-projection. Thus, `sum(a * b)`, given `a * b => x`, is translated as `sum(x)`.

> I'm confused about the semantics of having an extra `columnReferences` argument in the `put` method.